flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -335,15 +336,51 @@ public void scheduleOnlyIfVertexIsCreated() throws Exception {
 		}
 	}
 
+	@Test
+	public void vertexIsNotAffectedByOutdatedDeployment() {
+		final JobGraph jobGraph = singleNonParallelJobVertexJobGraph(2);
+
+		testExecutionSlotAllocator.disableAutoCompletePendingRequests();
+		final DefaultScheduler scheduler = createSchedulerAndStartScheduling(jobGraph);
+
+		final Iterator<ArchivedExecutionVertex> vertexIterator = scheduler.requestJob().getAllExecutionVertices().iterator();
+		final ArchivedExecutionVertex v1 = vertexIterator.next();
+		final ArchivedExecutionVertex v2 = vertexIterator.next();
+
+		final SchedulingExecutionVertex sv1 = scheduler.getSchedulingTopology().getVertices().iterator().next();
+
+		// fail v1 and let it recover to SCHEDULED
+		scheduler.updateTaskExecutionState(new TaskExecutionState(
+			jobGraph.getJobID(),
+			v1.getCurrentExecutionAttempt().getAttemptId(),
+			ExecutionState.FAILED));
+		taskRestartExecutor.triggerScheduledTasks();
+
+		// fail v2 to finish all pending slot requests of the last round
```

Comment:
I don't understand this comment or the test. `v1` and `v2` are completely independent from each other since there is only one job vertex in this job.

Reply:
Ok, understood.